workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/ScreenOverlayDialogFactory.kt

```diff
@@ -17,74 +18,7 @@ import com.squareup.workflow1.ui.startShowing
- * the main UI.
- *
- *    +-------------------------+
- *    |                         |
- *    |  BodyAndOverlaysScreen  |
- *    |                         |
- *    +-------------------------+
- *    |    TutorialBarScreen    |
- *    +-------------------------+
- *
- * Suppose we have custom dialogs that we want to cover the entire screen, except when
- * tutorial is running -- the tutorial bar should always be visible, and should always
- * be able to field touch events.
- *
- * To support this case we provide the [OverlayArea] value in the [ViewEnvironment].
- * When a [BodyAndOverlaysScreen] includes [overlays][BodyAndOverlaysScreen.overlays],
- * the [OverlayArea] holds the bounds of the view created to display the
- * [body screen][BodyAndOverlaysScreen.body]. Well behaved dialogs created to
- * display those [Overlay] renderings look for [OverlayArea] value and restrict
- * themselves to the reported bounds.
- *
- * Another [ViewEnvironment] value is maintained to support modality: [CoveredByModal].
- * When this value is true, it indicates that a dialog window driven by a [ModalOverlay]
- * is in play over the view, or is about to be, and so touch and click events should be
- * ignored. This is necessary because there is a long period after a call to
- * [Dialog.show][android.app.Dialog.show] before the new Dialog window will start
- * intercepting events -- an issue that has preoccupied Square for a
- * [very long time](https://stackoverflow.com/questions/2886407/dealing-with-rapid-tapping-on-buttons).)
- *
- * The default container view driven by [BodyAndOverlaysScreen] automatically honors
- * [CoveredByModal]. Dialog windows built by [ScreenOverlayDialogFactory] also honor
- * [CoveredByModal] when there is a [ModalOverlay]-driven dialog with a higher Z index in play.
- * All of this is driven by the [LayeredDialogSessions] support class, which can also be used to
- * create custom [BodyAndOverlaysScreen] container views. To put such a custom container
- * in play, see [OverlayDialogFactoryFinder].
- *
- * Modality also has implications for the handling of the Android back button.
- * While a modal dialog is shown,
- * [back press handlers][com.squareup.workflow1.ui.backPressedHandler] on covered
- * views and windows should not fire. [ScreenOverlayDialogFactory] takes care
- * of that requirement by default, presuming that your app uses Jetpack
- * [OnBackPressedDispatcher][androidx.activity.OnBackPressedDispatcher]. If that is not
- * the case, override [buildDialogWithContent] and provide an alternative `onBackPressed`
- * implementation when you call [OverlayDialogHolder].
- *
- * It is important to note that the modal behavior described here is all keyed to the
- * [ModalOverlay] interface, not its parent type [Overlay]. Rendering types that declare the
- * latter but not the former can be used to create dialogs for non-modal windows like toasts
- * and tool tips.
- */
+@Deprecated("Use ComponentDialog.setContent")
```

Comment:
Remember if you do change the name of `setContent` I don't think it will auto-refactor these deprecation warning strings?

Reply:
I had indeed forgotten, thanks.